python/pyspark/sql/utils.py

```diff
@@ -73,39 +74,6 @@ def __init__(
             self.cause = convert_exception(origin.getCause())
         self._origin = origin
 
-    def __str__(self) -> str:
```

Comment:
I think it might make sense to move this back. The captured exception is actually thrown from the JVM and thus the place where a JVM backtrace is actually present. So it much rather belongs here than in the parent class.